ext/mixed/js/core.js

```diff
@@ -288,6 +293,45 @@ const yomichan = (() => {
             this.trigger('orphaned', {error});
         }
 
+        getTemporaryListenerResult(eventHandler, userCallback, timeout=null) {
+            let resolve;
+            let reject;
+            const listenerPromise = new Promise((_resolve, _reject) => {
+                resolve = _resolve;
+                reject = _reject;
+            });
+
+            if (eventHandler === chrome.runtime.onMessage) {
+                const runtimeMessageCallback = ({action, params}, sender, sendResponse) => {
+                    let timeoutId = null;
+                    if (timeout !== null) {
+                        timeoutId = window.setTimeout(() => {
+                            timeoutId = null;
+                            reject(new Error(`Listener timed out in ${timeout} ms`));
```

Comment:
Move `reject` after `eventHandler.removeListener`; looks more return-like that way.

Reply:
https://github.com/FooSoft/yomichan/pull/417/commits/8c32ce7799fac5dc0694ab51ab65d31c3ea76fe1